### PR TITLE
Check disk by mount point threshold

### PR DESCRIPTION
**Pull request by Xiaohua-Shen**

### CODE REVIEW

File: roles/common/tasks/monitoring.yml [hunk 1]
```diff
@@ -146,8 +146,16 @@
   sensu_check: name=network_interface_traffic plugin=check-netif.rb state=absent
   notify: restart sensu-client
 
-- name: disk check
-  sensu_check: name=disk plugin=check-disk.rb
+- name: disk check default
```
Comment (j2sol) [inline roles/common/tasks/monitoring.yml]: I'm confused by this name. Looking at the arguments I'm guessing you're checking disk levels for everything except `/opt/stack/data`.

But then later you have a specific check for `/`, which is already being checked here, so I'm confused as to the desired outcome.

File: roles/percona-server/tasks/monitoring.yml [hunk 2]
```diff
@@ -16,3 +16,8 @@
   notify: restart collectd
   when: collectd is defined and collectd.enabled|bool
   tags: collectd
+
+- name: disk check for /var/lib/mysql
+  sensu_check: name=disk-mysql plugin=check-disk-by-mount.rb args='-m /var/lib/mysql -w 70 -c 80'
```
Comment (j2sol) [inline roles/percona-server/tasks/monitoring.yml]: Given that we're already checking all paths except `/opt/stack/data` wouldn't this lead to double alerts should `/var/lib/mysql` fill up?

File: roles/common/tasks/monitoring.yml (hunk 1 above)
Comment (Xiaohua-Shen) [inline roles/common/tasks/monitoring.yml]: I will modify the code to ignore / and /var/lib/mysql in default disk check.

File: roles/percona-server/tasks/monitoring.yml (hunk 2 above)
Comment (Xiaohua-Shen) [inline roles/percona-server/tasks/monitoring.yml]: I will modify the code to ignore / and /var/lib/mysql in default disk check.

File: roles/common/tasks/monitoring.yml [hunk 3]
```diff
@@ -38,6 +38,10 @@
             owner=root group=sensu mode=0640
   notify: restart sensu-client
   when: monitoring.client_key is defined
+  
+- name: install dependancy gems
+  shell: /opt/sensu/embedded/bin/gem install {{ item.name }} --no-user-install
+  with_items: {{monitoring.checks.gems}}
```
Comment (j2sol) [inline roles/common/tasks/monitoring.yml]: Please use a space between brackets and words: `{{ monitoring.checks.gems }}`

File: roles/monitoring-common/defaults/main.yml [hunk 4]
```diff
@@ -10,6 +10,15 @@ monitoring:
     cpu:
       warning: 80
       critical: 99
+    disk:
+      criticality: 'critical'
+      ignore_mnts_check: '/opt/stack/data,/,/var/lib/mysql'
+      specific_mnts_check:
+        - { name: 'stackdata', path: '/opt/stack/data', warning: 89, critical: 99 }
+        - { name: 'root', path: '/', warning: 80, critical: 90 }
+        - { name: 'mysql', path: '/var/lib/mysql', warning: 70, critical: 80 }
```
Comment (j2sol) [inline roles/monitoring-common/defaults/main.yml]: It's a bit more readable with this format:
```
specific_mnts_check:
  - name: 'stack data'
    path: '/opt/stack/data'
    warning: 89
    critical: 99
  - name: root
    path: '/'
```

File: roles/monitoring-common/defaults/main.yml [hunk 5]
```diff
@@ -10,6 +10,15 @@ monitoring:
...
+    gems:
+      - { name: 'sys-filesystem' }
```
Comment (j2sol) [inline roles/monitoring-common/defaults/main.yml]: Shouldn't need to make this a dict, just make it a simple list.

**Comment by j2sol** [comment]
ok to test

**Comment by j2sol** [comment]
add to whitelist

**Comment by Xiaohua-Shen** [comment]
retest

**Pull request merged**